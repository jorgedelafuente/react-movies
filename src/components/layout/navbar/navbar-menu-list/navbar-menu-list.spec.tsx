import { render } from '@testing-library/react';
import NavbarMenuList from './navbar-menu-list';
import { RouterProvider, createRouter } from '@tanstack/react-router';
import { QueryClient } from '@tanstack/react-query';

import { routeTree } from '@/routeTree.gen';

const queryClient = new QueryClient();
const router = createRouter({
   routeTree,
   context: {
      queryClient,
   },
   defaultPreload: 'intent',
   defaultStaleTime: 60000,
});

describe('NavbarMenuList', () => {
   it('NavbarMenuList should match snapshot', () => {
      const { container } = render(
         <RouterProvider router={router} defaultComponent={NavbarMenuList} />
      );
      expect(container).toMatchSnapshot();
   });
});
