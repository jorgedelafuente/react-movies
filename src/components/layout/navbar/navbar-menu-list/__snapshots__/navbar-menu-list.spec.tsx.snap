// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavbarMenuList > NavbarMenuList should match snapshot 1`] = `
<div>
  <div>
    <button>
      retry
    </button>
    <div
      style="padding: .5rem; max-width: 100%;"
    >
      <div
        style="display: flex; align-items: center; gap: .5rem;"
      >
        <strong
          style="font-size: 1rem;"
        >
          Something went wrong!
        </strong>
        <button
          style="appearance: none; font-size: .6em; border: 1px solid currentcolor; padding: .1rem .2rem; font-weight: bold; border-radius: .25rem;"
        >
          Hide Error
        </button>
      </div>
      <div
        style="height: .25rem;"
      />
      <div>
        <pre
          style="font-size: .7em; border: 1px solid red; border-radius: .25rem; padding: .3rem; color: red; overflow: auto;"
        >
          <code>
            No QueryClient set, use QueryClientProvider to set one
          </code>
        </pre>
      </div>
    </div>
  </div>
</div>
`;
